### DISCUSSION TITLE
[Enhancements] Caching Implementations

### DISCUSSION THREAD

**Discussion by TropicLegend** [Ani-Moopa/Moopa#72] [Ideas]
## What is the feature you'd like?

I would like caching to be implemented.

Please describe the feature you'd like to see in this project.

Please implement some forms of caching. Either with redis or maybe just file-caches. An optional opt-in Feature which can be enabled in the .env.

## Why do you think this feature would be useful?

It would greatly benefit page loading times.

Please explain why you think this feature would be useful.

More people would like to watch anime on a speedy website.

## Are there any alternatives you've considered?

General optimisation of code would also be an option, but I think caching would actually be easier to implement and generally quite faster. It would also help the server to not have as much load need it doesn't have to repeat operations it has already done before.

**Comment by DevanAbinaya** [comment]
Absolutely, I already have caching for API that revalidate each 10 minutes, I might be increasing the time but I'm afraid that it could be a problem for on-going anime, where when new episode come out and it still used cached data since the time hasn't run out. 

Thank you for the suggestions, just like you I also want to make it fast and figuring the best way to optimized code and stuff but that would take me longer cause there's so much stuff that I have to learn first 😅

**Comment by TropicLegend** [comment]
Is it maybe possible to cache parts of the web pages? So maybe not the episode stuff but rather the page itself? 

Unfortunately I have never done any coding with nodejs, so I don't know which options there are.

API calls might not even be the big problem but rather - for example, the homepage load times. I think it would greatly benefit the server performance and load times if the home page (except ofc. the "currently watching" tab) was cached.

If there is any way, shape or form I can help with testing, please let me know.

I got to say though, the page itself looks really awesome and works very well. If loading times would be just a tad faster, it'd be perfect ;).

(Currently on vacation, will be back next Monday)

**Comment by DevanAbinaya** [comment]
hi, I've implemented caching using redis to the site but idk if I do it correctly, but it's really improve page load by a lot just like you said. Thank you so much for suggesting this :)
